### ISSUE TITLE
Enable wireguard on 5.9.y kernel

### ISSUE THREAD

**Issue by gongsearch** [raspberrypi/linux#3945]
Hey together,

could you please consider enabling wireguard in 5.9.y?

This would be great.

Thanks
Sven

**Comment by ghost** [comment]
Per https://github.com/raspberrypi/linux/pull/3557, it's already enabled on arm64 builds, but perhaps not on the 5.9 branch.

**Comment by Neustradamus** [comment]
Before the end of this year, WireGuard will be in the next LTS "5.10.x" branch.

**Comment by pelwell** [comment]
Wireguard is enabled for arm64 kernels on rpi-5.9.y and rpi-5.10.y.

**Comment by MichaIng** [comment]
I opened a PR to add it for armhf as well, hope I did that right: #3967

**Comment by AlphaJack** [comment]
@pelwell Why not on armv7?

**Comment by pelwell** [comment]
As stated in #3555 and #3557:
```
I see you've enabled it on 64-bit only, which ought to make any overhead in size or performance less noticeable.
```
All 64-bit capable platforms are guaranteed to have at least 1GB of RAM and a faster processor.

**Comment by MichaIng** [comment]
See my PR which got already merged, i.e. the current 5.10 branch contains the WireGuard module for ARMv7/armhf.

Not sure what the performance and CPU argument should be, since the alternative OpenVPN is much more inefficient than WireGuard and there is really zero problem to run WireGuard even on RPi Zero 🙂.

**Comment by AlphaJack** [comment]
Indeed, using the DKMS module on a similar device with 100Mb/s ethernet gives me 95-85Mb/s through Wireguard, absolutely without increasing CPU or RAM usage.

**Comment by WertiaCoffee** [comment]
Same issue here. Concerned about enabling wireguard in 5.9y

**Comment by pelwell** [comment]
Work on 5.9 stopped months ago - 5.10 is an LTS kernel and will be released to the stable branch any day now.
